core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -500,7 +500,10 @@ int HIP::impl_is_initialized() {
 }
 
 void HIP::impl_initialize(InitializationSettings const& settings) {
-  Impl::HIPInternal::singleton().initialize(::Kokkos::Impl::get_gpu(settings));
+  hipStream_t singleton_stream;
+  hipStreamCreate(&singleton_stream);
```

Comment:
Hmm maybe a name like `base_stream`, `general_stream` or `default_stream` is more appropriate?

Reply:
IMHO, any of these suggestions could imply that this should be the default stream which is precisely what we don't want. Also, we are initializing the singleton here.

Reply:
Makes sense.